rust_src/src/cmds.rs

```diff
@@ -9,3 +9,19 @@ pub fn forward_point(n: LispObject) -> LispObject {
     let pt = ThreadState::current_buffer().pt();
     LispObject::from_fixnum(n.as_fixnum_or_error() + pt as EmacsInt)
 }
+
+#[lisp_fn(min = "0")]
+pub fn beginning_of_line(n: LispObject) -> LispObject {
+
```

Comment:
I think it would look better if you make n mutable and avoid an additional variable and the else block.

Reply:
The C code is modifying a local copy. That is what this code is explicitly doing. Adding the `mut` keyword makes reading the function signature more complex because now I have to look, does it take a reference? Why are they just mutating a copy? Is LispObject now a type wrapper around a reference? All of that is not work saving the creation of a local variable in my opinion. Or is code golf the intention here?

Reply:
I have noticed some other spots in our codebase, where it was done this way. It's not important to me how we handle it, but maybe we should discuss a guide line for this in order to create some consistency.

Reply:
Sounds good.